addons/pos_sale/static/src/overrides/models/pos_store.js

```diff
@@ -150,12 +150,12 @@ patch(PosStore.prototype, {
             const product_unit = line.product_id.uom_id;
             if (product_unit && !product_unit.is_pos_groupable) {
                 let remaining_quantity = newLine.qty;
+                newLine.delete();
```

Comment:
Why this change ?

Reply:
Otherwise there is a bug with remaining_quantity, that lead to creation of useless lines